merlin/study/celerymanageradapter.py

```diff
@@ -0,0 +1,126 @@
+    for worker in workers:
+        if redis_connection.exists(worker[0]):
+            redis_connection.hset(worker[0], "monitored", 1)
+            redis_connection.hset(worker[0], "pid", worker[1])
+        worker_info = WORKER_INFO
+        worker_info["pid"] = worker[1]
+        redis_connection.hmset(name=worker[0], mapping=worker_info)
+    redis_connection.quit()
+
+
+def remove_monitor_workers(workers: list):
+    """
+    Remove workers from being monitored by the celery manager.
+    :param list workers:        A worker names
+    """
+    if workers is None or len(workers) <= 0:
+        return
+    redis_connection = CeleryManager.get_worker_status_redis_connection()
+    for worker in workers:
+        if redis_connection.exists(worker):
+            redis_connection.hset(worker, "monitored", 0)
+            redis_connection.hset(worker, "status", WorkerStatus.stopped)
+
+    redis_connection.quit()
+
+
+def is_manager_runnning() -> bool:
+    """
+    Check to see if the manager is running
+
+    :return: True if manager is running and False if not.
+    """
+    redis_connection = CeleryManager.get_worker_args_redis_connection()
+    manager_status = redis_connection.hgetall("manager")
+    redis_connection.quit()
+    return manager_status["status"] == WorkerStatus.running and psutil.pid_exists(manager_status["pid"])
+
+
+def run_manager(query_frequency: int = 60, query_timeout: float = 0.5, worker_timeout: int = 180) -> bool:
+    """
+    A process locking function that calls the celery manager with proper arguments.
+    :params:        See CeleryManager for more information regarding the parameters
+    """
+    celerymanager = CeleryManager(query_frequency=query_frequency, query_timeout=query_timeout, worker_timeout=worker_timeout)
+    celerymanager.run()
+
+
+def start_manager(query_frequency: int = 60, query_timeout: float = 0.5, worker_timeout: int = 180) -> bool:
+    """
+    A Non-locking function that calls the celery manager with proper arguments.
+    :params:        See CeleryManager for more information regarding the parameters
+
+    :return bool:       True if the manager was started successfully.
+    """
+    subprocess.Popen(
+        f"merlin manager run -qf {query_frequency} -qt {query_timeout} -wt {worker_timeout}",
+        shell=True,
+        close_fds=True,
+        stdout=subprocess.PIPE,
+    )
+    return True
+
+
+def stop_manager() -> bool:
+    """
+    Stop the manager process using it's pid.
+
+    :return bool:       True if the manager was stopped successfully and False otherwise.
+    """
+    redis_connection = CeleryManager.get_worker_status_redis_connection()
+    manager_pid = int(redis_connection.hget("manager", "pid"))
+    manager_status = redis_connection.hget("manager", "status")
+    print(redis_connection.hgetall("manager"))
+    redis_connection.quit()
+
+    print(manager_status, psutil.pid_exists(manager_pid))
```

Comment:
Should these be LOG statements instead of prints?

Reply:
@bgunnar5 Corrected print statements to LOG statements.